### PR TITLE
port Httpreporters to httpclientservice

### PR DESCRIPTION
**Pull request by tardyp**

### CODE REVIEW

File: master/buildbot/reporters/bitbucket.py [hunk 1]
```diff
@@ -47,13 +49,26 @@ def reconfigService(self, oauth_key, oauth_secret,
...
     def send(self, build):
         results = build['results']
+        oauth_request = yield self.oauthhttp.post("",
+                                                  json=_GET_TOKEN_DATA)
+        if oauth_request.code == 200:
+            content_json = yield oauth_request.json()
+            token = content_json['access_token']
+        else:
+            content = yield oauth_request.content()
+            log.error("{}: unable to authenticate to Bitbucket {}".format(
```
Comment (sa2ajj) [inline master/buildbot/reporters/bitbucket.py]: should this also be `.err` like above?

File: master/buildbot/reporters/bitbucket.py [hunk 2]
```diff
@@ -47,13 +49,26 @@ def reconfigService(self, oauth_key, oauth_secret,
         if base_url.endswith('/'):
             base_url = base_url[:-1]
 
-        self._base_url = base_url
-        self._oauth_url = oauth_url
-        self._auth = (oauth_key, oauth_secret)
+        self.http = yield httpclientservice.HTTPClientService.getService(
```
Comment (sa2ajj) [inline master/buildbot/reporters/bitbucket.py]: do we need to make it public (i.e. use `_http` instead)?

File: master/buildbot/reporters/bitbucket.py [hunk 3]
```diff
@@ -47,13 +49,26 @@ def reconfigService(self, oauth_key, oauth_secret,
...
+            self.master, base_url)
+
+        self.oauthhttp = yield httpclientservice.HTTPClientService.getService(
```
Comment (sa2ajj) [inline master/buildbot/reporters/bitbucket.py]: ditto

**Comment by sa2ajj** [comment]
This change is worth a mention in release notes.

**Comment by tardyp** [comment]
@sa2ajj thanks for the review. I added releases notes, and made the http attribute private in the lasts commits.

File: master/buildbot/reporters/bitbucket.py (hunk 1 above)
Comment (tardyp) [inline master/buildbot/reporters/bitbucket.py]: this module uses the new logger api, which log.error is an indication of loglevel

File: master/buildbot/changes/gitpoller.py [hunk 4]
```diff
@@ -315,7 +315,7 @@ def _process_changes(self, newRev, branch):
             failures = [r[1] for r in results if not r[0]]
             if failures:
                 for failure in failures:
-                    log.error(
+                    log.err(
```
Comment (tardyp) [inline master/buildbot/changes/gitpoller.py]: I found this one via a search. looks like this line is not covered :-/

**Pull request merged**